reproject/mosaicking/coadd.py

```diff
@@ -106,6 +121,14 @@ def reproject_and_coadd(
 
     wcs_out, shape_out = parse_output_projection(output_projection, shape_out=shape_out)
 
+    if final_array is not None and final_array.shape != shape_out:
+        raise ValueError("If you specify an output array, it must have a shape matching "
+                         "the output shape")
+    if final_footprint is not None and final_footprint.shape != shape_out:
+        raise ValueError("If you specify an output footprint array, it must have a shape matching "
+                         "the output shape")
```

Comment:
Could you re-write these error messages to mention the expected and actual shape in each case?

Reply:
done